File: frontend/src/types/account.ts
export interface Account {
  account_id: string;
  customer_id: string;
  name: string;
  account_type: string;
  bank: string;
  country: string;
  city: string;
  created_at: string;
  status: 'ACTIVE' | 'SUSPENDED' | 'FROZEN' | 'DORMANT';
  risk_score: number;
  risk_level: 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL';
  device_id: string;
  phone: string;
  email: string;
}

export interface AccountDetailResponse {
  account: Account;
  risk_score: number;
  risk_level: string;
  detected_patterns: string[];
  evidence: string[];
  stats: {
    transaction_count: number;
    incoming_volume: number;
    outgoing_volume: number;
    connected_nodes_count: number;
  };
  recent_transactions: any[];
  mini_graph: {
    nodes: any[];
    edges: any[];
  };
}

File: backend/app/models/account.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime

class Account(BaseModel):
    account_id: str
    customer_id: str
    name: str
    account_type: str = "SAVINGS"  # SAVINGS, CURRENT, BUSINESS, DIGITAL_WALLET
    bank: str
    country: str = "India"
    city: str = "Mumbai"
    created_at: str
    status: str = "ACTIVE"  # ACTIVE, SUSPENDED, FROZEN, DORMANT
    risk_score: float = 0.0
    risk_level: str = "LOW"  # LOW, MEDIUM, HIGH, CRITICAL
    device_id: str
    phone: str
    email: str

class AccountCreate(BaseModel):
    account_id: str
    customer_id: str
    name: str
    account_type: str = "SAVINGS"
    bank: str
    country: str = "India"
    city: str = "Mumbai"
    created_at: Optional[str] = None
    status: str = "ACTIVE"
    risk_score: float = 0.0
    risk_level: str = "LOW"
    device_id: str
    phone: str
    email: str
